app/src/main/java/org/fossasia/phimpme/share/SharingActivity.java

```diff
@@ -1081,107 +1079,95 @@ void uploadImgur() {
     Bitmap bitmap = getBitmapFromPath(saveFilePath);
     final String imageString = getStringImage(bitmap);
     // sending image to server
-    StringRequest request =
-        new StringRequest(
-            Request.Method.POST,
-            Constants.IMGUR_IMAGE_UPLOAD_URL,
-            new Response.Listener<String>() {
+    ApiInterface retrofitClient =
+        RetrofitClient.getRetrofitClient(Urls.IMGUR_BASE_URL).create(ApiInterface.class);
```

Comment:
This should be done once and stored in a static variable

Reply:
We can't because we need to have retrofit client with multiple base Url @iamareebjamal

Reply:
> We can't because we need to have retrofit client with multiple base Url

Look at this, again
`RetrofitClient.getRetrofitClient(Urls.IMGUR_BASE_URL)`

Then look at this, again
`RetrofitClient.getRetrofitClient(Urls.IMGUR_BASE_URL).create(ApiInterface.class)`

And then read your comment

Reply:
@iamareebjamal I have created client for Imgur but when I want to create for any other api I have to pass that api base url, for example pinterest.

We can't create a singleton class and store in static variable when we have multiple base url.

Reply:
You can never use `ApiInterface` for any other base URL or even any other endpoint. If you can, I'll change my name to whatever you say

Reply:
https://stackoverflow.com/questions/41672427/how-to-make-a-singleton-for-retrofit-2

I'll recommend you do some research before making wrong claims

Reply:
> Look at this, again
RetrofitClient.getRetrofitClient(Urls.IMGUR_BASE_URL)
Then look at this, again
RetrofitClient.getRetrofitClient(Urls.IMGUR_BASE_URL).create(ApiInterface.class)

I will have another retrofit client for uploading image to Pinterest, that's why i have followed this approach. PR is in pending state due to failing volley multipart request.

Reply:
@iamareebjamal I know that we can create retrofit instance in a singleton class but this can be done when we have baseUrl only 1 through out the app. But as the pineterest account uses another base url I will have to create another singleton class for that base url as you suggested..

what i followed to not follow singleton approach and build the retrofit client with the base URL provided in the function itself.

```ApiInterface``` is just an interface with different endpoints. You can invoke that function which corresponds to that baseUrl. It makes sense that with Imgur base URL i will not call uploadImagetoPinterest function and vice versa.


If i am wrong please guide me what shall I do.?

Reply:
You want me to have multiple retrofit client in the app and mutiple interfaces for endpoints, I will go with this approach..

Reply:
Retrofit instance creation uses reflection and is expensive and should be done only once per application or activity, not on each function call

Reply:
@iamareebjamal Shall I create the retrofit instance in the activity itself and should I pass the URL from here or keep that where the retrofit instance is created?

Reply:
Pass from here